packages/ogmios/src/Provider/TxSubmitProvider/OgmiosTxSubmitProvider.ts

```diff
@@ -63,10 +66,28 @@ export class OgmiosTxSubmitProvider extends RunnableModule implements TxSubmitPr
     }
   }
 
-  async submitTx({ signedTransaction }: SubmitTxArgs): Promise<void> {
+  async submitTx({ signedTransaction, context }: SubmitTxArgs, handleProvider?: HandleProvider): Promise<void> {
     if (this.state !== 'running') {
       throw new CardanoNodeErrors.NotInitializedError('submitTx', this.name);
     }
+
+    if (context && context.handles && context.handles.length > 0) {
+      if (!handleProvider) {
```

Comment:
Better to handle this in the constructor

Reply:
Well, the handleProvider is passed into the `submitTx`, not into the constructor. And it's not an error until I'm using handles.. so I thought this the best way to pass it for now. 
If we want to pass it to the constructor and down to `submitTx` then I change how things are instantiated.

Reply:
Agree with @rhyslbw here: let's remove it from the core interface and add as a dependency in OgmiosTxSubmitProvider constructor